### DISCUSSION TITLE
File transfer QOL

### DISCUSSION THREAD

**Discussion by hannsen** [rustdesk/rustdesk#9118] [Feature Request]
First of all thanks for this great app!

I have some suggestions for file transfer because it can be a bit of a pain when dealing with slow connections.

I understand that file transfer is not the main part of the functionality so if there are any alternative softwares that specialize in easy file transfer (that work behind NAT) I would love to know of them.

So the suggestions are:
- show ETA for transfer completion
- allow to stop and resume file transfer
  - sometimes the speed slows to a crawl, but when I restart the file transfer it is faster again (but then I lose all the progress)
  - so it would be great if I could cancel the transfer and it would keep the partial .download file; and if I start the transfer again it can resume with the progress it had before
- general speed improvement tweaks
  - here I am not sure what can be done, but the transfer rate is usually well below the upload speed of the client
  - if I send the file over bittorrent it is considerably faster, but also a huge hassle to create the torrent first and add it on both sides, also one side has to have a port open
  - maybe there could be an option to send over multiple parallel connections or a different protocol, but I don't really have experience with that, I only know it could be faster

**Comment by jaloji** [comment]
Also if the connection is really bad, have a save of the queue to resume would be a great improvement.
